iocore/net/OCSPStapling.cc

```diff
@@ -207,17 +215,20 @@ ssl_stapling_init_cert(SSL_CTX *ctx, X509 *cert, const char *certname)
   if (cinf) {
     OPENSSL_free(cinf);
   }
+  if (map) {
+    delete map;
+  }
   return false;
 }
 
-static certinfo *
+static certinfo_map *
 stapling_get_cert_info(SSL_CTX *ctx)
 {
-  certinfo *cinf;
+  certinfo_map *map;
 
-  cinf = (certinfo *)SSL_CTX_get_ex_data(ctx, ssl_stapling_index);
-  if (cinf && cinf->cid) {
-    return cinf;
+  map = static_cast<certinfo_map *>(SSL_CTX_get_ex_data(ctx, ssl_stapling_index));
+  if (map && !map->empty() && map->begin()->second && map->begin()->second->cid) {
```

Comment:
Why does this check only the first element? Is there an ordering constraint that makes the appropriate element first?

Reply:
I see. That's really non-obvious. Perhaps an explanatory comment is in order?

Reply:
I'm just trying to be consistent with the original check.  If there are no entries, we cannot get any cert info.